packages/scaffold/src/views/w3m-networks-view/index.ts

```diff
@@ -97,6 +97,9 @@ export class W3mNetworksView extends LitElement {
         RouterController.push('SwitchNetwork', { ...data, network })
       }
     } else if (!isConnected) {
+      if (network.id.toLocaleLowerCase().includes('solana')) {
+        await NetworkController.switchActiveNetwork(network)
+      }
       NetworkController.setCaipNetwork(network)
```

Comment:
@svenvoskamp could we just use `switchActiveNetwork`  here? wyt? 
I think this might be some current redundant code.

Reply:
We don't want library-specific code within our scaffold package (solana check here). What is the problem you are running into? @valentyn-solana @gooxpf

Reply:
You shouldn't need `switchActiveNetwork` here, because you're not connected yet.